src/main/scala/com/ironcorelabs/AssociationType.scala

```diff
@@ -0,0 +1,24 @@
+package com.ironcorelabs.scala.sdk
+
+import com.ironcorelabs.{sdk => jsdk}
+
+/** Represents the reason a document can be viewed by the requesting user. */
+sealed trait AssociationType
```

Comment:
If you wouldn't mind, we should add `extends Product with Serializable` to the sealed traits we have. 

https://typelevel.org/blog/2018/05/09/product-with-serializable.html

I haven't done that consistently, but it's a good habit.

Reply:
Added here and `trait UserOrGroupId`